core/templates/pages/exploration-editor-page/services/user-exploration-permissions.service.ts

```diff
@@ -35,10 +35,15 @@ export class UserExplorationPermissionsService {
 
   static permissionsPromise: Promise<ExplorationPermissions> = null;
 
-  async getPermissionsAsync(): Promise<ExplorationPermissions> {
+  async getPermissionsAsync(refresh:boolean = false):
+  Promise<ExplorationPermissions> {
     if (!UserExplorationPermissionsService.permissionsPromise) {
       UserExplorationPermissionsService.permissionsPromise = (
         this.explorationPermissionsBackendApiService.getPermissions());
+    } else if (refresh) {
+      UserExplorationPermissionsService.permissionsPromise = (
+        this.explorationPermissionsBackendApiService.getPermissions());
+      return;
```

Comment:
Don't do it this way, have a separate function `flushPermissionsPromise` which will set permissionsPromise to null.

Reply:
DONE

Reply:
Sorry, I thought about this again and I think we can have a separate fetchPermissionsAsync function that will fetch the permission irrespective of whether we have any cached data.

Reply:
DONE